QUARKUS-803.md

```diff
@@ -0,0 +1,40 @@
+# QUARKUS-803 - Test and verify Red Hat DataGrid as part of the the RHBQ QE process
+
+JIRA link: https://issues.redhat.com/browse/QUARKUS-803
+
+Red Hat Data Grid is in-memory, distributed, NoSQL solution. It provides faster access to the data, then traditional relational database solution, trhough caching, near caching
+and operates in-memory rather than in computer storage. Users can benefit from this by storing in cache commonly accessed data that does not change frequently.
+Worth weighing rationality of using in-memory data grid as it introduces additional complexity to the system. The rule of thumb can be the number of frequent request made 
+to the service, if this number is high and queries to the database are complex (multiple joins and conditions) then it can speed up the system by caching these results.
+
+
+## Scope of the testing
```

Comment:
What about adding these scenarios:
- Failover scenarios: when one Quarkus application is connected to RHDG and the RHDG node gets down. The Quarkus application should not use the cache as RHDB is unavailable and then reconnect to the new RHDG node when is up.
- Data eviction after some time. This is another classic use case: when users want data to be in cache for some time only. When the evict time is out, the Quarkus application should evict the existing data in cache and update it accordingly.

Reply:
Very good reference Post about user cases: https://developers.redhat.com/articles/2021/05/28/building-real-time-leaderboard-red-hat-data-grid-and-quarkus-hybrid-kubernetes

Reply:
Failover scenarios are covered in OpenShift TS, they need to be migrated, that's covered imho in the TP.
https://github.com/quarkus-qe/quarkus-openshift-test-suite/tree/main/infinispan-client

Data eviction .. isn't that a feature of the datagrid itself, not a feature of the integration layer on Quarkus side ?

Reply:
Regarding data eviction, what is stated by Jose is more as a TTL time on an entry. Data eviction happening when there is not enough space/memory to store new cache entries and it should be a feature of datagrid server itself. 
I will add it to the test plan, it is a good use case